kornia/augmentation/functional.py

```diff
@@ -73,6 +74,29 @@ def random_grayscale(input: torch.Tensor, p: float = 0.5, return_transform: bool
     return apply_grayscale(input, params, return_transform)
 
 
+def random_perspective(input: torch.Tensor,
+                       distortion_scale: float = 0.5,
+                       p: float = 0.5,
+                       return_transform: bool = False) -> UnionType:
+    r"""Performs Perspective transformation of the given torch.Tensor randomly with a given probability.
+
+    Args:
+        input (torch.Tensor): Tensor to be transformed with shape (*, C, H, W).
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        distortion_scale(float): it controls the degree of distortion and ranges from 0 to 1. Default value is 0.5.
+        return_transform (bool): if ``True`` return the matrix describing the transformation
+        applied to each. Default: False.
+        input tensor.
+    """
+    if isinstance(input, tuple):
+        batch_size = input[0].shape[0] if len(input[0].shape) == 4 else 1
+    else:
```

Comment:
I think the input here is always a tensor, or at least it should be

Reply:
yes and no, if we want this function to behave as the module, it might receive a tuple. If that's the case, the type is wrong in the signature.

Reply:
Do we want that? I dont see a reason to make this function receive a tuple since its just a function and not an nn.Module that can be placed inside an nn.Sequential() in order to concatenate transforms. I think its better if we make the input here just a tensor and stick to just applying the transformation and returning the transformation matrix if specified.

Reply:
sounds good to me. Will adapt according to this behaviour. Adding to this PR random affine, so not merge it yet